### PR TITLE
Feature/secure element ci

### PR DESCRIPTION
**Pull request by Jagadish-NXP**
#### Problem
Secure Element build needs to be added in CI for K32W
we did observe that the crypto files were moving ahed with changes but the changes were not reflected ni the Secure Element build as the Secure Element Build was not part of CI. Adding it to CI will provide us immediate feedback when something is broken. Also now take care that there is no net compile time increase as suggested by andy31415

#### Change overview
Added Secure Element build for K32W

#### Testing
Manually tested on K32W061 with Secure Element

### CODE REVIEW

**Comment by andy31415** [comment]
Could you expand a bit more on the "Secure Element build needs to be added in CI for K32W" description?

We are starting to be sensitive on the time our build takes and how long CI takes to finish. Adding more platforms should be  done if they provide significant differences from  the rest. Does the secure element differ significantly from the regular one?

At the very least, could we compile just one of the lock or light app instead of both (and maybe delete one of them from the 'regular' build so that there is no net compile time increase?

**Review by andy31415 — CHANGES_REQUESTED** [review]
Requesting changes: concerned about extra build times adding up at only partial incremental gain.

Suggest to drop one of the non-secure-element builds and replace with se build, like "ligthing app uses regular build and lock app uses se build".

File: src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp [hunk 1]
```diff
@@ -500,7 +496,8 @@ CHIP_ERROR Spake2pHSM_P256_SHA256_HKDF_HMAC::ComputeRoundTwo(const uint8_t * in,
 
 CHIP_ERROR Spake2pHSM_P256_SHA256_HKDF_HMAC::KeyConfirm(const uint8_t * in, size_t in_len)
 {
-    VerifyOrExit(state == CHIP_SPAKE2P_STATE::R2, error = CHIP_ERROR_INTERNAL);
+    CHIP_ERROR error = CHIP_ERROR_INTERNAL;
```
Comment (andy31415) [inline src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp:499]: move declaration where it is used  rather than on top (need to put this on line 519). We want to decrease variable scope as much as possible. Previous version of the code was correct.

**Comment by Jagadish-NXP** [comment]
> Requesting changes: concerned about extra build times adding up at only partial incremental gain.
> 
> Suggest to drop one of the non-secure-element builds and replace with se build, like "ligthing app uses regular build and lock app uses se build".

Sure .. Update as suggested.. Now lock-app without SE and lighting app with SE to maintain the same executino time

File: src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp (hunk 1 above)
Comment (Jagadish-NXP) [reply src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp]: Sure. Done as Suggested

File: src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp [hunk 2]
```diff
@@ -487,7 +483,7 @@ CHIP_ERROR Spake2pHSM_P256_SHA256_HKDF_HMAC::ComputeRoundTwo(const uint8_t * in,
     uint8_t pKeyKe[16] = {
         0,
     };
-    constexpr size_t pkeyKeLen = sizeof(pKeyKe);
+    size_t pkeyKeLen = sizeof(pKeyKe);
```
[the remark this answers is not in the document]
Comment (Jagadish-NXP) [reply src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp:486]: mismatch when pkeyKeLen passed to next function hence constexpr removed

File: src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp [hunk 3]
```diff
@@ -518,8 +514,8 @@ CHIP_ERROR Spake2pHSM_P256_SHA256_HKDF_HMAC::KeyConfirm(const uint8_t * in, size
     {
         return Spake2p::KeyConfirm(in, in_len);
     }
-
-    const CHIP_ERROR error = Spake2p_KeyConfirm_HSM(&hsm_pake_context, role, in, in_len);
+    CHIP_ERROR error = CHIP_ERROR_INTERNAL;
+    error            = Spake2p_KeyConfirm_HSM(&hsm_pake_context, role, in, in_len);
```
Comment (bzbarsky-apple) [inline src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp:518]: Why is this change needed?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp (hunk 3 above)
Comment (bzbarsky-apple) [reply src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp]: And in particular, why can't `error` stay const here?

Comment (Jagadish-NXP) [reply src/crypto/hsm/nxp/CHIPCryptoPALHsm_SE05X_Spake2p.cpp]: It is the same now.. After the latest commit. This had got changed for VerifyOrExit which was being used sometime earlier and "error variable" was being reused. now VerifyOrReturnError is being used and the change is not required.

**Review by msandstedt — APPROVED** [review]

**Comment by woody-apple** [comment]
@Damian-Nordic @jepenven-silabs ?

**Review by andy31415 — APPROVED** [review]
